nixos/modules/installer/netboot/netboot.nix

```diff
@@ -141,6 +141,17 @@ with lib;
         # /etc/NIXOS tag.
         touch /etc/NIXOS
         ${config.nix.package}/bin/nix-env -p /nix/var/nix/profiles/system --set /run/current-system
+
+        # Set password for user nixos if specified on cmdline
+        # Allows using nixos-anywhere in headless environments
+        for o in $(</proc/cmdline); do
+          case "$o" in
+            live.nixos.passwd=*)
```

Comment:
I was thinking about how to make this a bit more secure, netboot often is not encrypted and while we cannot protect against active attacker trying to intercept/change the traffic, we shouldn't have a service where we can just ask for the password using the netboot protocol. This is a bit different from the ISO usecase which is just local and requires physical access.
So our perl script currently supports setting password via password hashes like this:

```
foreach my $u (values %usersOut) {
    next if defined $shadowSeen{$u->{name}};
    my $hashedPassword = "!";
    $hashedPassword = $u->{hashedPassword} if defined $u->{hashedPassword};
    my $expires = "";
    $expires = dateToDays($u->{expires}) if defined $u->{expires};
    # FIXME: set correct value for sp_lstchg.
    push @shadowNew, join(":", $u->{name}, $hashedPassword, "1::::", $expires, "") . "\n";
}
```

Could we change this bash instead or awk and do something similar?

This is how to generate a password hash:

```
echo foo | mkpasswd -m sha-512 -s
```

Reply:
There is a difference between an attacker that can ask the PXE server for the same configuration and an attacker that pretends to be the PXE server (active).

Reply:
You said you add the option, but I can't see it. Did you forgot to push?

Reply:
An attacker altering the response still makes no difference if it's a password or a hash. Needless to say an attacker in the position to run a pxe server (he needs control over the dhcp server as well for that) doesn't need a nixos netboot image to do harm. But I get your point with the cleartext password in the nix store. 

No, not forgotten. Just adding the test scenario. Pushing my current state though for your feedback.